### ISSUE TITLE
Support ESM plugins

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#7327]
### What is the problem you're trying to solve?

Currently, we support a plugin module only for CommonJS (CJS). It would be useful if we could support ESM as well.

https://github.com/stylelint/stylelint/blob/7cd8d00705e81cbe69bdbac85fd038cb26290e20/lib/augmentConfig.mjs#L323

### What solution would you like to see?

In addition to CJS, we should be able to import a plugin written in ESM. Probably, dynamic import (`import()`) may help implement it.

**Comment by wojtekLs** [comment]
Due to shitty nature of `js`, `cjs` &N `mjs` cumports SIM

**Comment by firefoxic** [comment]
When I first started developing the plugin, I decided that the v1.0.0 release would be with the release of Stylelint@16. And at that time it was not decided to support both module systems. So I already then made the plugin on ESM, and added babel to the build as a temporary patch. In v1.0.0 I planned to get rid of the build (and later try to get rid of jest, but not about that now).

It came as a surprise to me that the dual module system doesn't extend to plugin APIs right now, and only CJS are supported. It wouldn't be fun at all to drag the CJS build system from Stylelint into a plugin now. I think it kills the whole point of the period with dual modularity before switching to pure ESM.

Hope to implement support for ESM plugins 🙏
